test/v200/utils/common/command_test_utils.go

```diff
@@ -0,0 +1,212 @@
+package common
+
+import (
+	"fmt"
+
+	schema "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+)
+
+// commandAdded adds a new command to the test schema data and notifies the follower
+func (testDevFile *TestDevfile) commandAdded(command schema.Command) {
+	LogInfoMessage(fmt.Sprintf("command added Id: %s", command.Id))
+	testDevFile.SchemaDevFile.Commands = append(testDevFile.SchemaDevFile.Commands, command)
+	if testDevFile.Follower != nil {
+		testDevFile.Follower.AddCommand(command)
+	}
+}
+
+// commandUpdated and notifies the follower of the command which has been updated
+func (testDevFile *TestDevfile) commandUpdated(command schema.Command) {
+	LogInfoMessage(fmt.Sprintf("command updated Id: %s", command.Id))
+	if testDevFile.Follower != nil {
+		testDevFile.Follower.UpdateCommand(command)
+	}
+}
+
+// addEnv creates and returns a specifed number of env attributes in a schema structure
+func addEnv(numEnv int) []schema.EnvVar {
+	commandEnvs := make([]schema.EnvVar, numEnv)
+	for i := 0; i < numEnv; i++ {
+		commandEnvs[i].Name = "Name_" + GetRandomString(5, false)
+		commandEnvs[i].Value = "Value_" + GetRandomString(5, false)
+		LogInfoMessage(fmt.Sprintf("Add Env: %s", commandEnvs[i]))
+	}
+	return commandEnvs
+}
+
+// addAttributes creates returns a specifed number of attributes in a schema structure
+func addAttributes(numAtrributes int) map[string]string {
+	attributes := make(map[string]string)
+	for i := 0; i < numAtrributes; i++ {
```

Comment:
I did not see the `i` defined here being used. 
you should use `for key, value := range attributes`  to iterate through the map

Reply:
I am creating a map, not iterating through it, so not sure how range applies here.

Reply:
thanks for the explanation, I did not check the loop clearer